flagscale/backends/Megatron-LM/megatron/training/arguments.py.patch

```diff
@@ -339,11 +340,18 @@ index 1120c7529..190fac52b 100644
 +        assert args.num_experts is None, "PEFT is not tested with MoE currently"
 +        assert args.recompute_method is None and args.recompute_granularity is None and args.recompute_num_layers is None, "PEFT will raise comfilcts with recompute currently"
 +        assert args.ckpt_format == 'torch', "PEFT is only tested with torch format checkpoint"
++
++    if args.use_transformer_engine_fl:
++        assert not args.add_bias_linear, "Do not support linear with bias in TransformerEngine-TL Now"
++        assert args.context_parallel_size == 1, "DotProductAttention in TransformerEngine-FL, do not support context parallel now"
++        assert not args.tp_comm_overlap, "Do not support gemm/sp comm in TransformerEngine-FL now"
++        args.distributed_backend = 'flagcx'
++        args.use_flag_gems = True
```

Comment:
add logging to let users know that `use_transformer_engine_fl` is True, `flagcx` and `gems` also be used.